File: RegexFileSorter/Configuration.vb
Imports System.IO
Imports Newtonsoft.Json

Public Class Configuration
	Const CurrentF = "Current.json", ConfigsF = "Configs.json"

	Shared Sub New()
		Current = New Config
		Configs = New Dictionary(Of String, Config) From {{"C1", New Config}, {"C2", New Config}, {"C3", New Config}}
	End Sub

	Public Shared Property Current As Config
	Public Shared Property Configs As Dictionary(Of String, Config)

	Public Shared Sub LoadJSON()
		If File.Exists(CurrentF) Then
			Current = JsonConvert.DeserializeObject(Of Config)(File.ReadAllText(CurrentF))
		Else
			Current = New Config
		End If
		If File.Exists(ConfigsF) Then
			Configs = JsonConvert.DeserializeObject(Of Dictionary(Of String, Config))(File.ReadAllText(ConfigsF))
		Else
			Configs = New Dictionary(Of String, Config)
		End If
		'Configs.Add("Test", New Config())
		'Configs = JsonConvert.DeserializeObject(Of Dictionary(Of String, Config))(My.Settings.JSONConfigs)
	End Sub

	Public Shared Sub SaveJSON()
		File.WriteAllText(CurrentF, JsonConvert.SerializeObject(Current))
		File.WriteAllText(ConfigsF, JsonConvert.SerializeObject(Configs))
		'My.Settings.JSONConfig = JsonConvert.SerializeObject(Current)
		'My.Settings.JSONConfigs = JsonConvert.SerializeObject(Configs)
	End Sub

	Public Shared Sub Delete(Name As String)
		Configs.Remove(Name)
	End Sub

	Public Shared Sub SaveAs(Name As String)
		Configs.Add(Name, New Config(Current))
	End Sub

	Public Shared Sub SetConfig(Name As String)
		Current = New Config(Configs(Name))
	End Sub

End Class

File: RegexFileSorter/Configuration.cs
using Newtonsoft.Json;
using System.Collections.Generic;
using System.IO;

namespace RegexFileSorter
{
    internal static class Configuration
    {
        private const string CurrentF = "Current.json", ConfigsF = "Configs.json";

        static Configuration()
        {
            Current = new();
            Configs = new() { { "Default", new() } };
        }

        public static Dictionary<string, Config> Configs { get; set; }
        public static Config Current { get; set; }

        public static bool Contains(string name) => Configs.ContainsKey(name);

        public static void Delete(string name) => Configs.Remove(name);

        public static void LoadJSON()
        {
            if (File.Exists(CurrentF)) { Current = JsonConvert.DeserializeObject<Config>(File.ReadAllText(CurrentF)); }
            if (File.Exists(ConfigsF)) { Configs = JsonConvert.DeserializeObject<Dictionary<string, Config>>(File.ReadAllText(ConfigsF)); }
        }

        public static void SaveAs(string name) => Configs[name] = new Config(Current);

        public static void SaveJSON()
        {
            File.WriteAllText(CurrentF, JsonConvert.SerializeObject(Current));
            File.WriteAllText(ConfigsF, JsonConvert.SerializeObject(Configs));
        }

        public static void SetConfig(string name) => Current = new Config(Configs[name]);
    }
}